### PR TITLE
New package: zzz-user-hooks-1.0.0

### PR DESCRIPTION
**Pull request by anjandev**

### CODE REVIEW

File: srcpkgs/zzz-user-hooks/template [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+revision=1
+build_style=gnu-makefile
+checkdepends="shellcheck"
+short_desc="Allow user scripts for logged in user using zzz suspend/resume"
+maintainer="Anjandev Momi <anjan@momi.ca>"
+license="MIT"
+homepage="https://www.daveeddy.com/2018/10/01/run-user-scripts-on-suspend-and-wakeup-on-void-linux/"
+distfiles="https://github.com/bahamas10/zzz-user-hooks/archive/v${version}.tar.gz"
+checksum=bf29baf9a5690dafbede199f0a9d841759de828cd0967c3fe376e1155ab5c8fc
+ archs=noarch
```
Comment (sgn) [inline srcpkgs/zzz-user-hooks/template:13]: Do not indent here. :)

File: srcpkgs/zzz-user-hooks/template [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+do_install() {
+	vmkdir /etc/zzz.d/resume
+	vmkdir /etc/zzz.d/suspend
+	vcopy user-script /etc/zzz.d
+	vcopy hooks/resume/99-user-script /etc/zzz.d/resume
+	vcopy hooks/suspend/99-user-script /etc/zzz.d/suspend
```
Comment (sgn) [inline srcpkgs/zzz-user-hooks/template:20]: Use `vinstall` instead. And merge `post_install` into `do_install`

File: srcpkgs/zzz-user-hooks/template [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Template file for 'zzz-user-hooks'
+pkgname=zzz-user-hooks
+version=1.0.0
+revision=1
+build_style=gnu-makefile
+checkdepends="shellcheck"
```
Comment (sgn) [inline srcpkgs/zzz-user-hooks/template:6]: `depends="bash sudo"`

---
Below are not comment on template, but the code itself

- `bash`: It looks like they're using non-required bashism. With a quick skim over the code, the code could be rewritten in POSIX sh.
- `sudo`: I prefer `runuser` or `chpst` instead.

Comment (anjandev) [reply srcpkgs/zzz-user-hooks/template:6]: I agree with the bashisms but it seems the maintainer likes bashisms according to his blog.

I have opened an issue asking to change to runuser or chpst (https://github.com/bahamas10/zzz-user-hooks/issues/5).

File: srcpkgs/zzz-user-hooks/template [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Template file for 'zzz-user-hooks'
+pkgname=zzz-user-hooks
+version=1.0.0
+revision=1
+build_style=gnu-makefile
+depends="bash sudo"
```
Comment (ahesford) [inline srcpkgs/zzz-user-hooks/template:6]: This can be bumpbed to 1.1.0 and the `sudo` dependency removed, right?

**Comment by anjandev** [comment]
Hey,

Andrew J. Hesford <notifications@github.com> writes:

> This can be bumpbed to 1.1.0 and the `sudo` dependency removed, right?

Sorry I have been busy. I updated to 1.1.0 and removed sudo as a
dependency. Please merge this if you think it's ready.

> The X11 and Wayland detection in the upstream project is a little hacky, but that mean it isn't useful or suitable for packaging.

What would be the ideal way to detect X11 or Wayland? I can notify the
maintainer of the better way.

Thanks,
Anjandev Momi

**Comment by ahesford** [comment]
I'm not sure of the best way to detect Wayland or X11. In my own use of `zzz` hooks, I avoid the issue altogether by creating triggers like `/run/zzz/will-sleep` and touch the trigger file on suspend. Any X11 or Wayland program that needs to run can just watch the trigger with inotify and act when appropriate.

**Comment by sgn** [comment]
I guess no more complaints for this package?

**Comment by ahesford** [comment]
None from me

**Pull request merged**